tests/integration-all/api-gateway-websocket/tests.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const path = require('path');
+const AWS = require('aws-sdk');
+const _ = require('lodash');
+const { expect } = require('chai');
+
+const { getTmpDirPath, readYamlFile, writeYamlFile } = require('../../utils/fs');
+const {
+  region,
+  // confirmCloudWatchLogs,
+  createTestService,
+  deployService,
+  removeService,
+} = require('../../utils/misc');
+const {
+  createApi,
+  deleteApi,
+  getRoutes,
+  createStage,
+  deleteStage,
+} = require('../../utils/api-gateway-v2');
+
+const CF = new AWS.CloudFormation({ region });
+
+describe('AWS - API Gateway Websocket Integration Test', function() {
+  this.timeout(1000 * 60 * 10); // Involves time-taking deploys
+  let serviceName;
+  let stackName;
+  let tmpDirPath;
+  let serverlessFilePath;
+  let websocketApiId;
+  const stage = 'dev';
+
+  before(() => {
+    tmpDirPath = getTmpDirPath();
+    console.info(`Temporary path: ${tmpDirPath}`);
+    serverlessFilePath = path.join(tmpDirPath, 'serverless.yml');
+    const serverlessConfig = createTestService(tmpDirPath, {
+      templateDir: path.join(__dirname, 'service'),
+    });
+    serviceName = serverlessConfig.service;
+    stackName = `${serviceName}-${stage}`;
+    console.info(`Deploying "${stackName}" service...`);
+    deployService(tmpDirPath);
+    // create an external websocket API
+    const externalWebsocketApiName = `${stage}-${serviceName}-ext-api`;
+    return createApi(externalWebsocketApiName).then(wsApiMeta => {
+      websocketApiId = wsApiMeta.ApiId;
+      return createStage(websocketApiId, 'dev');
+    });
+  });
+
+  after(() => {
+    // NOTE: deleting the references to the old, external websocket API
+    const serverless = readYamlFile(serverlessFilePath);
+    delete serverless.provider.apiGateway.websocketApiId;
+    writeYamlFile(serverlessFilePath, serverless);
+    // NOTE: we need to delete the stage before deleting the stack
+    // otherwise CF will refuse to delete the deployment because a stage refers to that
+    return deleteStage(websocketApiId, 'dev').then(() => {
```

Comment:
As a note, altough we don't do that exensively, it's fine to rely on async/await in integration tests. We run them only in Node.js v12  (Goal is to test integration with external service, and not Node.js versions support in scope of a Framework)

Reply:
done